tests/preload/CMakeLists.txt

```diff
@@ -38,6 +38,11 @@ add_cstyle(tests-preload-config ${CMAKE_CURRENT_SOURCE_DIR}/config/config.c)
 add_check_whitespace(tests-preload-basic ${CMAKE_CURRENT_SOURCE_DIR}/basic/basic.c)
 add_check_whitespace(tests-preload-config ${CMAKE_CURRENT_SOURCE_DIR}/config/config.c)
 
+add_library(setumask SHARED setumask.c)
+set_target_properties(setumask PROPERTIES
+	INCLUDE_DIRECTORIES ${CMAKE_SOURCE_DIR}/src/
+	POSITION_INDEPENDENT_CODE ON)
```

Comment:
That travis build doesn't correspond to what is here, that still contains the extra OBJECT target.
```
Scanning dependencies of target setumask_o
[ 61%] Building C object tests/preload/CMakeFiles/setumask_o.dir/setumask.c.o
[ 61%] Built target setumask_o
...
[ 71%] Linking C shared library libsetumask.so
/usr/bin/ld: CMakeFiles/setumask_o.dir/setumask.c.o: relocation R_X86_64_PC32 against undefined symbol `umask@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
```
So of course, while compiling the object target, cmake didn't know if that should be PIE or not.